### ISSUE TITLE
UDM Pro 1.9.3.3488 not showing data under USW panel

### ISSUE THREAD

**Issue by bbagaria** [unifi-poller/unifi-poller#354]
First of all, thank you for the great work and the dashboard! 

I am a relatively new home pro user and have some issues (I think) which I could not locate in here.  Please delete or let me know if this is not the right forum.

I have UDM Pro v1.9.3.3488 while controller version 6.23.30; and with UAP-AC-PRO, USW-48-Port, UAP-AC-M & UAP-LR. I am using influx datasource with unifi-poller & grafna running on rpi4.

I am able to see UDM Pro data under USG Insights dashboard. This is good.
However, although I see UDM Pro listed under USW insights switch list but none of the panels show any data if I select UDM Pro.
Similarly, I see UAP-AC-Pro listed under USW insights switch list & it shows UAP-AC-Pro data in panels.
Additionally, I see USW-48 under USW insights switch list & it shows USW-48 data in panels.

Question, if I am reading this right.

USW dashboard is for switches and UAP-AC-Pro should not show (as it is not a switch) over there but UDM-Pro should as that is considered as a switch? If UAP-AC-Pro is indeed correct in USW insights, then should it not contain the UAP-AC-M, & UAP-LR as well (as both are of same family)?

The UAP insights dashboard does show all the UAP-AC-Pro and UAP-AC-M & UAP-LR data.

Any suggestions?

Thanks!

**Comment by davidnewhall** [comment]
Screenshots would help show me the problem you're reporting. You can also check the API output to see if the switch data you're expecting is even present. Running `unifi-poller -j d` will print out that data.

**Comment by bbagaria** [comment]
Hi,

 Here are some screenshots. Please let me know if you need any other.
1. UAP Insights showing : UAP-AC-Pro, UAP-AC-M and UAP-LR

![image](https://user-images.githubusercontent.com/10068320/119919626-6fcd6800-bf30-11eb-9cfb-61ec21f4754f.png)

2.USW Insight showing: UDM, UAP and USW and each device showing corresponding switch. Here as far as I understand UAP should not be listed as one of the devices as this is a access point and not a switch.

![image](https://user-images.githubusercontent.com/10068320/119919747-ac00c880-bf30-11eb-8595-03ad9d0d67ef.png)
![image](https://user-images.githubusercontent.com/10068320/119920859-a0160600-bf32-11eb-92c0-36b9e4ccf563.png)


3. USW Insight on selecting device as UDM and switch as UDM Pro, does not show any data in any of the graphs on this page. While this shows data correctly for US-48 switch. Oddly this also shows data for UAP (which is access point and not switch) but only for UAP-AC-Pro and not for UAP-AC-M (seems AC-M is not type of UAP in the poller data). UDM Pro instead should have the data as this is of category switch.
![image](https://user-images.githubusercontent.com/10068320/119919962-1dd91200-bf31-11eb-8a47-eaaea71cf1bf.png)
![image](https://user-images.githubusercontent.com/10068320/119920366-d901ab00-bf31-11eb-8efb-2ba837e59685.png)
![image](https://user-images.githubusercontent.com/10068320/119920616-401f5f80-bf32-11eb-8c95-2b8166332113.png)


4. USG Insight showing the right UDM Pro data as sent by poller
![image](https://user-images.githubusercontent.com/10068320/119920231-97710000-bf31-11eb-911f-696b8be5e156.png)

Output of unifi-poller -j d is giving me some 10K lines, so will go through them in a bit. I am wondering if UDM Pro is acting like USG (as that's where it showing the data) and not like USW (where its not showing the data) in this. Surely UDM Pro has temp, memory, cpu and load data. Naïve question, are you using snmp or APIs to get all the info from Unifi into timeseries db? I am trying to see if if I can get telegraf snmp to get the UDM Pro details into influxdb.

Thanks again!

**Comment by davidnewhall** [comment]
Poller uses the API. The `-j d` command you ran outputs all device data from the UniFi controller API. Use `jq` to manipulate it. The data is a large list of objects. You can select out just the UDM object.

UDMP devices show up in USG and USW dashboards. They have a switch and a security gateway built in. UDM (non-pro) also shows up on the UAP dashboard since that has a built-in WAP. Likewise, the UAP-AC-Pro has a built in switch (2 ports), so it also shows up in the USW dashboard along with a number of other UAPs w/ switch ports. I must assume the AC-M and UAP-LR devices you have only have 1 port and no switch.

My hunch is that rebooting your UDM will fix this. Sadly that has been the missing data fix most folks have come to need.

If that does fix it, you can compare the `-j d` output before/after to see what changed. If that does not fix it, we'll need to inspect that API output to figure out if the metrics are in a non-standard location. You're welcome to email it to me or DM me on Discord.

**Comment by bbagaria** [comment]
Apologize for the late response as I got busy at work and could not come back to this thread. I checked by a restart and that did not help. I haven't been able to look at the unifi-poller -j d output. Happy to send it to you in case it helps unifipoller.

**Comment by davidnewhall** [comment]
No worries. I need to look into this more. Thank you!

**Comment by Subtixx** [comment]
Same issue here, output of unifi-poller -j d shows the udmpro however:
```
"_id": "*ID*",
"ip": "*WAN IP*",
"mac": "*MAC ADDRESS*",
"model": "UDMPRO",
"type": "udm",
```

**Comment by bondskin** [comment]
> Same issue here, output of unifi-poller -j d shows the udmpro however:
> 
> ```
> "_id": "*ID*",
> "ip": "*WAN IP*",
> "mac": "*MAC ADDRESS*",
> "model": "UDMPRO",
> "type": "udm",
> ```
> 
> I cannot speak for anybody else, but in my case I also dont have a "udm" table in influx.
> Only the following:
> ![](https://camo.githubusercontent.com/b35eed08de21ef6272d88ad42ac1dc9f07af8ebb788ee62aad064c90ff527925/68747470733a2f2f692e696d6775722e636f6d2f7941526e6d30312e706e67)

Hi @Subtixx , for me, the UDM data shows up under "usg_*" tables

**Comment by Subtixx** [comment]
Yes @bondskin but looking at the queries of `Switch Temperatur` for example, it should have its own table =)

![image](https://user-images.githubusercontent.com/20743379/128630965-59a1c230-3b4e-4577-9f71-0b41bb884fb0.png)
